app/services/application_data_service.rb

```diff
@@ -0,0 +1,45 @@
+class ApplicationDataService
+  def self.gcse_qualifications_summary(application_form:)
+    gcses = %i[maths_gcse english_gcse science_gcse]
+    summary_string = gcses
+      .map { |gcse| application_form.send(gcse) }
+      .map { |gcse| summary_for_gcse(gcse) }
+      .compact
+      .join(',')
+
+    summary_string.presence
+  end
+
+  def self.missing_gcses_explanation(application_form:)
+    gcses = application_form.application_qualifications.select do |q|
+      q.level == 'gcse'
+    end
+
+    explanation = gcses
+                    .select(&:missing_qualification?)
+                    .map { |gcse| "#{gcse.subject.capitalize} GCSE or equivalent: #{gcse.missing_explanation}" }
+                    .join(',')
```

Comment:
The `join` has changed from newlines to a comma here.

- we probably ought to retain newlines in the API for consistency
- do we want something harder separating them, given explanations may contain commas